revocation/internal/x509util/validate.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package x509util provides the method to validate the certificate chain for a
+// specific purpose, including code signing and timestamping.
+package x509util
+
+import (
+	"crypto/x509"
+	"fmt"
+
+	"github.com/notaryproject/notation-core-go/revocation/purpose"
+	"github.com/notaryproject/notation-core-go/revocation/result"
+	coreX509 "github.com/notaryproject/notation-core-go/x509"
+)
+
+// ValidateChain checks the certificate chain for a specific purpose, including
+// code signing and timestamping.
+func ValidateChain(certChain []*x509.Certificate, certChainPurpose purpose.Purpose) error {
```

Comment:
nit: please create an issue for this: For revocation we shouldnt care if its time stamping or codesigning cert chain.,

Reply:
Created #227